### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by ultimatemonty**
Dockerizes the API

cc @pdougall1 @gschorkopf

### CODE REVIEW

**Comment by gschorkopf** [comment]
👏

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.git
+.env
+.dockerignore
```
Comment (gschorkopf) [inline .dockerignore]: I'm confused why `.dockerignore` is trying to ignore itself.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+root = true
```
Comment (gschorkopf) [inline .editorconfig]: While I agree with all of the decisions here, I'm not sure I understand why this has been committed.

File: README.md [hunk 3]
```diff
@@ -26,6 +26,33 @@ To run tests, run `rake db:test:prepare; rspec`.
...
+  volumes:
+    - .:/var/www/frolfr-server
+
+web:
+  # use whatever volumes are configured for the app container
+  volumes_from:
+    - app
+
+```
+This file is already in ,gitignore so if you want to make local tweaks go nuts!
```
Comment (gschorkopf) [inline README.md]: Typo `,gitignore`

**Comment by gschorkopf** [comment]
So (to be honest), before I continue reviewing, I think I need the ELI5 or even a ELI10 on why this helps `frolfr-server`. I've frankly never used Docker and have no idea how it applies to this API.

File: README.md [hunk 4]
```diff
@@ -26,6 +26,33 @@ To run tests, run `rake db:test:prepare; rspec`.
...
+If process isolation is your bad then you're in luck - the Frolfr API is full containerized! Make sure you have docker fully setup and running in your local environment and then do the following to get frolfr running:
+```
+# build images
+docker-compose build
+
+# start the containers
+docker-compose up
+```
+
+You can now access Frolfr at 0.0.0.0!
```
Comment (gschorkopf) [inline README.md]: Can't you already access `frolfr-server` if you just type `unicorn` or `rails s`? What does "access" mean?

File: README.md [hunk 5]
```diff
@@ -26,6 +26,33 @@ To run tests, run `rake db:test:prepare; rspec`.
 
 You'll need to configure cloudinary, [including adding a cloudinary.yml file](http://cloudinary.com/documentation/rails_integration).
 
+#### Running Frolfr in Docker
+If process isolation is your bad then you're in luck - the Frolfr API is full containerized! Make sure you have docker fully setup and running in your local environment and then do the following to get frolfr running:
```
Comment (gschorkopf) [inline README.md]: Process isolation *is* my bad, because I'm not sure what it means. Isolating client from server? Server from... the DB? Other things?

**Comment by ultimatemonty** [comment]
@gschorkopf typos abound :( I was honestly so excited it was working I didn't really slow down to check for typos.

Apologies for not providing a rationale in the original PR comment. The point of this is more for front-end development. Instead of having to setup Rails locally (which depending on platform can be a pain) you just setup docker and run the API that way. You should be able to develop against the dockerized API as well but I haven't tested it out yet.

What I have tested is running the commands in the README then running `ember server` from the client-app and watching things magically work without me ever needing to install Rails and deal with random environment stuff as has happened in the past for me (due to general lack of Ruby/Rails experience).

File: README.md (hunk 5 above)
Comment (ultimatemonty) [inline README.md]: supposed to be `bag`
source

File: .dockerignore (hunk 1 above)
Comment (ultimatemonty) [inline .dockerignore]: `.dockerignore` lists out files we don't want copied from the local filesystem to the dockerized app when building the containers. See [the docs](https://docs.docker.com/engine/reference/builder/#/dockerignore-file) for more info

**Comment by ultimatemonty** [comment]
@gschorkopf fixed typos and removed/.gitignored `.editorconfig`

File: .dockerignore (hunk 1 above)
Comment (gschorkopf) [inline .dockerignore]: Looked at the docs. Still don't know why the file `.dockerignore` includes the line `.dockerignore`.

**Comment by gschorkopf** [comment]
I'll be able to get to this tomorrow. Seems innocuous enough to ship as-is, but would like to poke around a bit.

**Comment by gschorkopf** [comment]
🚢
